audio-worklet/design-pattern/lib/wa2.js

```diff
@@ -0,0 +1,289 @@
+/**
+ * @fileOverview WA2 library. A set of utilities to support WASM operations
+ * for the WebAudio.
+ */
+
+
+/**
+ * @namespace WA2
+ */
+var WA2 = {
+  // Basic byte unit of WASM heap. (16 bit = 2 bytes)
+  BYTES_PER_UNIT: Uint16Array.BYTES_PER_ELEMENT,
+
+  // Byte per audio sample. (32 bit float)
+  BYTES_PER_SAMPLE: Float32Array.BYTES_PER_ELEMENT,
+
+  // The max audio channel on Chrome is 32.
+  MAX_CHANNEL_COUNT: 32,
+
+  // WebAudio's render quantum size.
+  RENDER_QUANTUM_FRAMES: 128,
+};
+
+
+/**
+ * Copy AudioBuffer by a specified channel count.
+ * @param  {AudioBuffer} sourceBuffer
+ * @param  {AudioBuffer} destinationBuffer
+ * @param  {number} channelCount A number of channels to be copied.
+ */
+WA2.copyAudioBuffer = function(sourceBuffer, destinationBuffer, channelCount) {
+  console.assert(channelCount <= sourceBuffer.numberOfChannels);
```

Comment:
console.assert or throw?  or maybe console.assert with a message.

Reply:
I am removing this function. This was needed for the ScriptProcessorNode.